qiskit_nature/circuit/library/ansatzes/utils/fermionic_excitation_generator.py

```diff
@@ -128,6 +124,56 @@ def generate_fermionic_excitations(
         The list of excitations encoded as tuples of tuples. Each tuple in the list is a pair. The
         first tuple contains the occupied spin orbital indices whereas the second one contains the
         indices of the unoccupied spin orbitals.
+
+    Examples:
+        Generate excitations with basic inputs.
+
+        .. jupyter-execute::
+
+            from qiskit_nature.circuit.library.ansatzes.utils.fermionic_excitation_generator import (
+                generate_fermionic_excitations
+            )
+
+            num_excitations = 1
+            num_spin_orbitals = 4
+            num_particles = [1,1]
+
+            excitations = generate_fermionic_excitations(
+                num_excitations, num_spin_orbitals, num_particles
+            )
+
+        Generate excitations limiting the maximum number of excitations per spin species.
+
+        .. jupyter-execute::
+
+            from qiskit_nature.circuit.library.ansatzes.utils.fermionic_excitation_generator import (
```

Comment:
If I recall correctly the `jupyter-execute` blocks are executed sequentially and context-aware. That means that the import is already present from a previously executed blocks so you can skip the import here and further down below :+1:

Reply:
In terms of jupyter-execute I know Terra is trying to limit their usage, see  Qiskit/qiskit-terra#7661

When I referred to including examples above I referred to doctest i.e. https://docs.python.org/3/library/doctest.html which we support here in CI and examples will be run and the output as stored checked. The code samples would need to produce output, for example the list of excitations produced

Reply:
@woodsp-ibm do you have any more specific doctest examples of how this is implemented in nature? I am struggling to get this to work, the following docstring doesn't seem to render any output when i build the docs:
```
.. testcode::
    from qiskit_nature.circuit.library.ansatzes.utils.fermionic_excitation_generator import (
           generate_fermionic_excitations
     )
            
      excitations = generate_fermionic_excitations(1, 4, (1, 1), generalized=True)
      print(excitations)
        
.. testoutput::
    [((0,), (1,)), ((2,), (3,))]
```

<img width="854" alt="Screenshot 2022-02-24 at 3 04 51 PM" src="https://user-images.githubusercontent.com/23662430/155599013-11b7ddee-a830-4f1b-8c15-314c56791059.png">

I also tried the following which just rendered a completeley empty box:
```
.. doctest::
    >>>excitations = generate_fermionic_excitations(1, 4, (1, 1), generalized=True)
    >>>print(excitations)
    [((0,), (1,)), ((2,), (3,))]
```

Reply:
As far as I am aware you just include it in the doctring. Here is an example from finance https://github.com/Qiskit/qiskit-finance/blob/5c9107ddd733e312569f3f2ef4d5a0cdfd19da36/qiskit_finance/circuit/library/probability_distributions/lognormal.py#L62-L70 (I don't see anything in Nature). Now looking at the output that results maybe it needs to be inside some markup https://qiskit.org/documentation/finance/stubs/qiskit_finance.circuit.library.LogNormalDistribution.html#qiskit_finance.circuit.library.LogNormalDistribution as it has rolled all the output into a single line @manoelmarques any thoughts here - does it need to be inside a code python block perhaps?

Reply:
I believe that doctest enforces a whitespace after `>>>`. See here: https://docs.python.org/3/library/doctest.html#how-it-works They don't explicitly state that but when explaining where the expected output should be formatted they explicitly write `>>> ` and `... ` including the trailing whitespace!